### DISCUSSION TITLE
how to  calculate nhance  1999-2000 dii？？

### DISCUSSION THREAD

**Discussion by peck82** [jamesjiadazhan/dietaryindex#212] [Q&A] [answered]
"I want to use the 1999-2000 data to calculate relevant indicators, but I have a few questions. First, I'm not sure if the Mypyr_tot.exe file I downloaded is correct. Second, is it possible to get feedback on any issues with the code and provide me with a correct R code that can run successfully?"
<error/rlang_error>
![image](https://github.com/jamesjiadazhan/dietaryindex/assets/135391805/9e2d5a9e-866d-4082-a7c1-7044c4f00f38)

library(devtools) #加载devtools包
install_github("jamesjiadazhan/dietaryindex")
library(dietaryindex)
library(haven)
DEMO_PATH = read_xpt(file ='DEMO.XPT')#人口
FPED_PATH_1 = read_sas('pyr_tot.sas7bdat')#第一天数据
NUTRIENT_PATH_1 = read_xpt(file ='DRXTOT.XPT')#第一天数据
#FPED_PATH：第一天fped数据；NUTRIENT_PATH：第一天营养数据；DEMO_PATH:人口统计数据；FPED_PATH2：第二天fped数据；NUTRIENT_PATH2：第二天营养数据。
DII_NHANES_FPED(FPED_PATH=FPED_PATH_1, NUTRIENT_PATH=NUTRIENT_PATH_1, DEMO_PATH=DEMO_PATH)
MED_NHANES_FPED(FPED_PATH=FPED_PATH_1, NUTRIENT_PATH=NUTRIENT_PATH_1, DEMO_PATH=DEMO_PATH)

Error in `filter()`:
ℹ In argument: `DR1DRSTZ == 1`.
Caused by error:
! 找不到对象'DR1DRSTZ'

**Comment by jamesjiadazhan** [comment]
Thanks for this question. The NHANES 1999-2005 functions are still under development. Once they are finished, they will be published. The next update will happen in the next 1-2 months. If you plan to publish manuscripts using the 1999-2005 data soon, you can reach out to my email at jzha832@emory.edu to have a discussion, as I can prioritize work.

**Comment by jamesjiadazhan** [comment]
I just added the DII and HEI2015 for 1999-2000 data. The function's name is DII_NHANES_MPED. Good luck.

**Reply by lizhigang123-tec** [reply]
Hello, Thank you from the bottom of my heart for sharing， but I couldn't find the function "DII_NHANES_MPED" after loading R package, please how to solve this problem!

**Reply by jamesjiadazhan** [reply, marked as answer]
Thanks for using dietaryindex!

Please first try to reinstall dietaryindex for the latest update. Although DII_NHANES_MPED is not mentioned in the manual, you should be able to use it directly after loading the package. I double-checked it and it’s there.
![IMG_5218](https://github.com/jamesjiadazhan/dietaryindex/assets/108076575/711ef580-00b5-45ac-a92f-e7c314349b33)

If you want to know the usage of DII_NHANES_MPED, use ?DII_NHANES_MPED.

Just a friendly reminder, if you use dietaryindex for publication, please give us a star on GitHub and cite our work.
- Jiada James Zhan, Rebecca A Hodge, Anne Dunlop, et al. Dietaryindex: A User-Friendly and Versatile R Package for Standardizing Dietary Pattern Analysis in Epidemiological and Clinical Studies. bioRxiv. Published online August 07, 2023:2023.08.07.548466. doi:10.1101/2023.08.07.548466
- https://www.biorxiv.org/content/10.1101/2023.08.07.548466v2

**Comment by zhaoweiezhaoweiba** [comment]
hi  about the DII-nhases-MEPD, where could i can download the WJFPT data?

**Comment by lizhigang123-tec** [comment]
Hello, you don't need to download WJFPT data by yourself. Just reinstall the "dietaryindex" package and use the "DII_NHANES_MPED" function can calculate the DII of 1999-2000






------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2023年12月14日(星期四) 晚上8:35
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [jamesjiadazhan/dietaryindex] how to  calculate nhance  1999-2000 dii？？ (Discussion #212)





 
hi  about the DII-nhases-MEPD, where could i can download the WJFPT data?

**Comment by zhaoweiezhaoweiba** [comment]
thank you for the reply!  the code below is OK for 2003-2004(WJFRT = NHANES_20032004$WJFRT was included in but no need for download? 


MPED_PER_100_GRAM_PATH=read_sas('pyr_tot_d1.sas7bdat')
#WJFRT=read_sas('wjfrt.sas7bdat')
NUTRIENT_PATH = read_xpt(file ='DR1TOT_C.XPT')
NUTRIENT_IND_PATH=read_xpt(file ='DR1IFF_C.XPT')
DEMO_PATH=read_xpt(file ='DEMO_C.XPT')
NUTRIENT_PATH2=read_xpt(file ='DR2TOT_C.XPT')
NUTRIENT_IND_PATH2 =read_xpt(file ='DR2IFF_C.XPT')
data("NHANES_20032004")
DII_C<-DII_NHANES_MPED(MPED_PER_100_GRAM_PATH = NHANES_20032004$MPED_PER_100_GRAM, WJFRT = NHANES_20032004$WJFRT, NUTRIENT_PATH = NHANES_20032004$NUTRIENT, NUTRIENT_IND_PATH = NHANES_20032004$NUTRIENT_IND, DEMO_PATH = NHANES_20032004$DEMO, NUTRIENT_PATH2 = NHANES_20032004$NUTRIENT2, NUTRIENT_IND_PATH2 = NHANES_20032004$NUTRIENT_IND2)

View(DII_C)

**Comment by zhaoweiezhaoweiba** [comment]
thank you for the reply!  the code below is OK for 2003-2004(WJFRT = NHANES_20032004$WJFRT was included in but no need for download? 




MPED_PER_100_GRAM_PATH=read_sas('pyr_tot_d1.sas7bdat')
#WJFRT=read_sas('wjfrt.sas7bdat')
NUTRIENT_PATH = read_xpt(file ='DR1TOT_C.XPT')
NUTRIENT_IND_PATH=read_xpt(file ='DR1IFF_C.XPT')
DEMO_PATH=read_xpt(file ='DEMO_C.XPT')
NUTRIENT_PATH2=read_xpt(file ='DR2TOT_C.XPT')
NUTRIENT_IND_PATH2 =read_xpt(file ='DR2IFF_C.XPT')
data("NHANES_20032004")
DII_C<-DII_NHANES_MPED(MPED_PER_100_GRAM_PATH = NHANES_20032004$MPED_PER_100_GRAM, WJFRT = NHANES_20032004$WJFRT, NUTRIENT_PATH = NHANES_20032004$NUTRIENT, NUTRIENT_IND_PATH = NHANES_20032004$NUTRIENT_IND, DEMO_PATH = NHANES_20032004$DEMO, NUTRIENT_PATH2 = NHANES_20032004$NUTRIENT2, NUTRIENT_IND_PATH2 = NHANES_20032004$NUTRIENT_IND2)


View(DII_C)










--

WEIE ZHAO
PHD


Reproductive Medicine Research Center 

Sixth Affiliated Hospital of Sun Yan-Sen University

17#, Sogoulin Rd Guangzhou 510080, China 510655

Tel: +8615889970468

***@***.***

http://www.newivf.org










At 2023-12-14 19:58:53, "lizhigang123-tec" ***@***.***> wrote:

Hello, you don't need to download WJFPT data by yourself. Just reinstall the "dietaryindex" package and use the "DII_NHANES_MPED" function can calculate the DII of 1999-2000






------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;;
发送时间: 2023年12月14日(星期四) 晚上8:35
收件人: ***@***.***&gt;;
抄送: ***@***.***&gt;; ***@***.***&gt;;
主题: Re: [jamesjiadazhan/dietaryindex] how to calculate nhance 1999-2000 dii？？ (Discussion #212)






hi about the DII-nhases-MEPD, where could i can download the WJFPT data?

**Comment by zhaoweiezhaoweiba** [comment]
thank you for the reply!  the code below is OK for 2003-2004(WJFRT = NHANES_20032004$WJFRT was included in but no need for download? 


MPED_PER_100_GRAM_PATH=read_sas('pyr_tot_d1.sas7bdat')
#WJFRT=read_sas('wjfrt.sas7bdat')
NUTRIENT_PATH = read_xpt(file ='DR1TOT_C.XPT')
NUTRIENT_IND_PATH=read_xpt(file ='DR1IFF_C.XPT')
DEMO_PATH=read_xpt(file ='DEMO_C.XPT')
NUTRIENT_PATH2=read_xpt(file ='DR2TOT_C.XPT')
NUTRIENT_IND_PATH2 =read_xpt(file ='DR2IFF_C.XPT')
data("NHANES_20032004")
DII_C<-DII_NHANES_MPED(MPED_PER_100_GRAM_PATH = NHANES_20032004$MPED_PER_100_GRAM, WJFRT = NHANES_20032004$WJFRT, NUTRIENT_PATH = NHANES_20032004$NUTRIENT, NUTRIENT_IND_PATH = NHANES_20032004$NUTRIENT_IND, DEMO_PATH = NHANES_20032004$DEMO, NUTRIENT_PATH2 = NHANES_20032004$NUTRIENT2, NUTRIENT_IND_PATH2 = NHANES_20032004$NUTRIENT_IND2)

View(DII_C)

**Reply by jamesjiadazhan** [reply]
Correct, sorry for my late reply

**Comment by lizhigang123-tec** [comment]
I am very sorry that I just rechecked the data and code, and it seems that "dietaryindex" only contains the data of 2017-2018. I only remember that there is a detailed record in a file on github, so I can download it from the official website as required.






------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2023年12月14日(星期四) 晚上9:09
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [jamesjiadazhan/dietaryindex] how to  calculate nhance  1999-2000 dii？？ (Discussion #212)





 

感谢您的回复！下面的代码在 2003-2004 年是可以的（WJFRT = NHANES_20032004$WJFRT 包含在其中，但不需要下载？
 

MPED_PER_100_GRAM_PATH=read_sas（'pyr_tot_d1.sas7bdat'）
 #WJFRT=read_sas('wjfrt.sas7bdat')

NUTRIENT_PATH = read_xpt（文件 ='DR1TOT_C.XPT'）

NUTRIENT_IND_PATH=read_xpt（文件 ='DR1IFF_C.XPT'）

DEMO_PATH=read_xpt（文件 ='DEMO_C.XPT'）

NUTRIENT_PATH2=read_xpt（文件 ='DR2TOT_C.XPT'）

NUTRIENT_IND_PATH2 =read_xpt（文件 ='DR2IFF_C.XPT'）
&nbsp;数据（“NHANES_20032004”）

DII_C<-DII_NHANES_MPED（MPED_PER_100_GRAM_PATH = NHANES_20032004$MPED_PER_100_GRAM，WJFRT = NHANES_20032004$WJFRT，NUTRIENT_PATH = NHANES_20032004$NUTRIENT，NUTRIENT_IND_PATH = NHANES_20032004$NUTRIENT_IND，DEMO_PATH = NHANES_20032004$DEMO，NUTRIENT_PATH2 = NHANES_20032004$NUTRIENT2，NUTRIENT_IND_PATH2 = NHANES_20032004$NUTRIENT_IND2）
 
&nbsp;查看（DII_C）
 
—

直接回复此邮件，在 GitHub 上查看，或取消订阅。

您收到此消息是因为您发表了评论。  Message ID: ***@***.***&gt;

**Reply by jamesjiadazhan** [reply]
Thanks for your help!

**Comment by jamesjiadazhan** [comment]
Hi, 

Please check the following instructions to download the compiled data from NHANES. I used to include all compiled data when downloading the package but GitHub prohibits the direct download when the size of package is too big, so I decided to include only the 2017-2018 data as a reference. You may download the data as shown below:

dietaryindex has compiled NHANES data from 1999 - 2020 for your convenience. This includes NHANES 1999-2002 (MPED was merged by USDA), 2003-2004, 2005-2006, 2007-2008, 2009-2010, 2011-2012, 2013-2014, 2015-2016, 2017-2018, 2017-2020. To retrieve the data:

Download the NHANES_combined folder from the Google Drive (https://drive.google.com/drive/u/2/folders/1umjhuS22aHEW_bU5AjYa8vrae91gsb0D)
Download the NHANES_combined folder from the GitHub dietaryindex_NHANES page (https://github.com/jamesjiadazhan/dietaryindex_NHANES/tree/main/data/NHANES_combined)

**Comment by zhaoweiezhaoweiba** [comment]
Thank you for you reply , you mean I can download the NHANES combined data and use  DII_NHANES_MPED.R for calucalation of DII?







--

WEIE ZHAO
PHD


Reproductive Medicine Research Center 

Sixth Affiliated Hospital of Sun Yan-Sen University

17#, Sogoulin Rd Guangzhou 510080, China 510655

Tel: +8615889970468

***@***.***

http://www.newivf.org










At 2023-12-14 21:34:07, "James Jiada Zhan" ***@***.***> wrote:

Hi,

Please check the following instructions to download the compiled data from NHANES. I used to include all compiled data when downloading the package but GitHub prohibits the direct download when the size of package is too big, so I decided to include only the 2017-2018 data as a reference. You may download the data as shown below:

dietaryindex has compiled NHANES data from 1999 - 2020 for your convenience. This includes NHANES 1999-2002 (MPED was merged by USDA), 2003-2004, 2005-2006, 2007-2008, 2009-2010, 2011-2012, 2013-2014, 2015-2016, 2017-2018, 2017-2020. To retrieve the data:

Download the NHANES_combined folder from the Google Drive (https://drive.google.com/drive/u/2/folders/1umjhuS22aHEW_bU5AjYa8vrae91gsb0D)
Download the NHANES_combined folder from the GitHub dietaryindex_NHANES page (https://github.com/jamesjiadazhan/dietaryindex_NHANES/tree/main/data/NHANES_combined)

**Reply by jamesjiadazhan** [reply]
yes

**Comment by zhaoweiezhaoweiba** [comment]
Regarding the years 2003-2004, using the following code yields results. However, for the years 1999-2002 (where only the first day's diet is available), an error occurs. Could you please help me verify if the results for 2003-2004 are correct, and advise on how to handle the data for the years 1999-2002?





load("NHANES_20032004.rda")

DII_NHANES_MPED(MPED_PER_100_GRAM_PATH = NHANES_20032004$MPED_PER_100_GRAM, 

                       WJFRT = NHANES_20032004$WJFRT, NUTRIENT_PATH = NHANES_20032004$NUTRIENT, 

                       NUTRIENT_IND_PATH = NHANES_20032004$NUTRIENT_IND, DEMO_PATH = NHANES_20032004$DEMO, 

                       NUTRIENT_PATH2 = NHANES_20032004$NUTRIENT2, NUTRIENT_IND_PATH2 = NHANES_20032004$NUTRIENT_IND2)













--

WEIE ZHAO
PHD


Reproductive Medicine Research Center 

Sixth Affiliated Hospital of Sun Yan-Sen University

17#, Sogoulin Rd Guangzhou 510080, China 510655

Tel: +8615889970468

***@***.***

http://www.newivf.org










At 2023-12-14 21:34:07, "James Jiada Zhan" ***@***.***> wrote:

Hi,

Please check the following instructions to download the compiled data from NHANES. I used to include all compiled data when downloading the package but GitHub prohibits the direct download when the size of package is too big, so I decided to include only the 2017-2018 data as a reference. You may download the data as shown below:

dietaryindex has compiled NHANES data from 1999 - 2020 for your convenience. This includes NHANES 1999-2002 (MPED was merged by USDA), 2003-2004, 2005-2006, 2007-2008, 2009-2010, 2011-2012, 2013-2014, 2015-2016, 2017-2018, 2017-2020. To retrieve the data:

Download the NHANES_combined folder from the Google Drive (https://drive.google.com/drive/u/2/folders/1umjhuS22aHEW_bU5AjYa8vrae91gsb0D)
Download the NHANES_combined folder from the GitHub dietaryindex_NHANES page (https://github.com/jamesjiadazhan/dietaryindex_NHANES/tree/main/data/NHANES_combined)

**Reply by jamesjiadazhan** [reply]
You're using the function for 2003-2004 correctly. Let me investigate what happened to NHANES 2001 to 2002

**Comment by zhaoweiezhaoweiba** [comment]
for NHANES 1999-2002，there is no DR1TP182










--

WEIE ZHAO
PHD


Reproductive Medicine Research Center 

Sixth Affiliated Hospital of Sun Yan-Sen University

17#, Sogoulin Rd Guangzhou 510080, China 510655

Tel: +8615889970468

***@***.***

http://www.newivf.org










在 2023-12-14 21:34:23，"James Jiada Zhan" ***@***.***> 写道：

Thanks for your help!

**Reply by lizhigang123-tec** [reply]
大佬，你是要算1999-2000的DII吗？

**Comment by zhaoweiezhaoweiba** [comment]
Thank you for the help！  Have you identified what the specific problem is for NHANES 1999-2002?













--

WEIE ZHAO
PHD


Reproductive Medicine Research Center 

Sixth Affiliated Hospital of Sun Yan-Sen University

17#, Sogoulin Rd Guangzhou 510080, China 510655

Tel: +8615889970468

***@***.***

http://www.newivf.org










At 2023-12-19 20:36:07, "James Jiada Zhan" ***@***.***> wrote:

You're using the function for 2003-2004 correctly. Let me investigate what happened to NHANES 2001 to 2002

**Reply by jamesjiadazhan** [reply]
I updated the codes. It should work now.

Let me know if you have other questions. If possible, please give our work a Star on GitHub and cite our work when you publish the work. It helps recognize and appreciate our intellectual efforts and contributions.

Jiada James Zhan, Rebecca A Hodge, Anne Dunlop, et al. Dietaryindex: A User-Friendly and Versatile R Package for Standardizing Dietary Pattern Analysis in Epidemiological and Clinical Studies. bioRxiv. Published online August 07, 2023:2023.08.07.548466. doi:10.1101/2023.08.07.548466

**Comment by zhaoweiezhaoweiba** [comment]
Thanks!

how to cite for DII_NHANES 1999-2002, "A User-Friendly and Versatile R Package for Standardizing Dietary Pattern Analysis in Epidemiological and Clinical Studies " is OK?










--

WEIE ZHAO
PHD


Reproductive Medicine Research Center 

Sixth Affiliated Hospital of Sun Yan-Sen University

17#, Sogoulin Rd Guangzhou 510080, China 510655

Tel: +8615889970468

***@***.***

http://www.newivf.org










At 2023-12-20 22:29:33, "James Jiada Zhan" ***@***.***> wrote:

I updated the codes. It should work now.

Let me know if you have other questions. If possible, please give our work a Star on GitHub and cite our work when you publish the work. It helps recognize and appreciate our intellectual efforts and contributions.

Jiada James Zhan, Rebecca A Hodge, Anne Dunlop, et al. Dietaryindex: A User-Friendly and Versatile R Package for Standardizing Dietary Pattern Analysis in Epidemiological and Clinical Studies. bioRxiv. Published online August 07, 2023:2023.08.07.548466. doi:10.1101/2023.08.07.548466

**Reply by jamesjiadazhan** [reply]
yes. Check it here: https://pubmed.ncbi.nlm.nih.gov/37609152/

**Comment by zhaoweiezhaoweiba** [comment]
是的，发布者大佬已经做出了这个功能













--

WEIE ZHAO
PHD


Reproductive Medicine Research Center 

Sixth Affiliated Hospital of Sun Yan-Sen University

17#, Sogoulin Rd Guangzhou 510080, China 510655

Tel: +8615889970468

***@***.***

http://www.newivf.org










在 2023-12-20 22:33:33，"lizhigang123-tec" ***@***.***> 写道：

大佬，你是要算1999-2000的DII吗？

**Reply by lizhigang123-tec** [reply]
之前不就已经有了么，我之前10月份的时候也问过他

**Comment by zhaoweiezhaoweiba** [comment]
我最近才尝试使用，多多指教！







--

WEIE ZHAO
PHD


Reproductive Medicine Research Center 

Sixth Affiliated Hospital of Sun Yan-Sen University

17#, Sogoulin Rd Guangzhou 510080, China 510655

Tel: +8615889970468

***@***.***

http://www.newivf.org










在 2023-12-20 22:36:03，"lizhigang123-tec" ***@***.***> 写道：

之前不久已经有了么，我之前10月份的时候也问过他

**Reply by lizhigang123-tec** [reply]
我刚才又去确认了一遍，还是可以算的。只是1999-2000和2001-2002的数据好像只能算第一天的，2003以后的就可以算2天的DII了。如果你需要帮助的话，我的邮箱是2570687492@qq.com

**Comment by zhaoweiezhaoweiba** [comment]
谢谢您的帮助，是的，现在可以算了，那我加您QQ可以吗













--

WEIE ZHAO
PHD


Reproductive Medicine Research Center 

Sixth Affiliated Hospital of Sun Yan-Sen University

17#, Sogoulin Rd Guangzhou 510080, China 510655

Tel: +8615889970468

***@***.***

http://www.newivf.org










在 2023-12-20 22:55:26，"lizhigang123-tec" ***@***.***> 写道：

***@***.***

**Comment by lizhigang123-tec** [comment]
可以的



发自我的iPhone


------------------ Original ------------------
From: zhaoweiezhaoweiba ***@***.***&gt;
Date: Wed,Dec 20,2023 11:58 PM
To: jamesjiadazhan/dietaryindex ***@***.***&gt;
Cc: lizhigang123-tec ***@***.***&gt;, Comment ***@***.***&gt;
Subject: Re: [jamesjiadazhan/dietaryindex] how to  calculate nhance  1999-2000 dii？？ (Discussion #212)





 谢谢您的帮助，是的，现在可以算了，那我加您QQ可以吗
 
 
 
 
 
 
 
 
 
 
 
 
 
 --
 
 WEIE ZHAO
 PHD
 
 
 Reproductive Medicine Research Center 
 
 Sixth Affiliated Hospital of Sun Yan-Sen University
 
 17#, Sogoulin Rd Guangzhou 510080, China 510655
 
 Tel: +8615889970468
 
 ***@***.***
 
 http://www.newivf.org
 
 
 
 
 
 
 
 
 
 
 在 2023-12-20 22:55:26，"lizhigang123-tec" ***@***.***&gt; 写道：
 
 ***@***.***
